server/src/main/java/org/opensearch/search/query_group/QueryGroupServiceSettings.java

```diff
@@ -0,0 +1,317 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.query_group;
+
+import org.opensearch.common.settings.ClusterSettings;
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.unit.TimeValue;
+
+/**
+ * Main class to declare the QueryGroup feature related settings
+ */
+public class QueryGroupServiceSettings {
```

Comment:
This should be separate PR for the settings IMO. Does not have anything to do with the CRUD API